### PR TITLE
OnDropFiles: Add support for spaces, quotes and backslahes in filename

### PR DESCRIPTION
**Pull request by mwestphal**
`OnDropFiles()` needs to escape quotes and backslashes in filenames before adding quotes around them.
Also `TriggerBinding()` should add a space between commands and first argument instead of expecting said space to be there already.

Test for spaces in filename is not supported by the interaction testing mechanism.
Test for quotes in filename is supported but github actions do not like quotes in filenames, so testing is limited.
Test for backslashes in filename is supported on Windows

Superseed https://github.com/f3d-app/f3d/pull/1725

### CODE REVIEW

**Comment by mwestphal** [comment]
@snoyer @Meakk please review

File: library/src/interactor_impl.cxx [hunk 1]
```diff
@@ -244,10 +245,19 @@ class interactor_impl::internals
   {
     internals* self = static_cast<internals*>(clientData);
     vtkStringArray* filesArr = static_cast<vtkStringArray*>(callData);
+    const std::regex charsToEscape(R"(([\"\\]))");
```
Comment (snoyer) [inline library/src/interactor_impl.cxx:248]:
```suggestion
    const std::regex charsToEscape(R"((["\\]))");
```
shouldn't you also drop the first backslash when going from `"([\"\\\\])"` to an `R"` string?

Comment (mwestphal) [reply library/src/interactor_impl.cxx:248]: absolutely !

File: library/src/utils.cxx [hunk 2]
```diff
@@ -31,7 +31,15 @@ std::vector<std::string> utils::tokenize(std::string_view str)
...
-        escaped = true;
+        if (escaped)
+        {
+          accumulate(c);
+          escaped = false;
+        }
+        else
+        {
+          escaped = true;
+        }
```
Comment (Meakk) [inline library/src/utils.cxx:42]:
```suggestion
        if (escaped)
        {
          accumulate(c);
        }
        escaped = !escaped;
```

**Review by Meakk — APPROVED** [review]
classic nitpicking comment, but otherwise lgtm

File: library/src/interactor_impl.cxx [hunk 3]
```diff
@@ -424,15 +434,21 @@ class interactor_impl::internals
...
           // XXX: Ignore the boolean return of triggerCommand,
           //  error is already logged by triggerCommand
-          this->Interactor.triggerCommand(command + argsString);
+          this->Interactor.triggerCommand(commandWithArgs);
         }
         catch (const f3d::interactor::command_runtime_exception& ex)
         {
-          log::error("Interaction: error running command:\"", command + argsString, "\":");
+          log::error("Interaction: error running command:\"", commandWithArgs, "\":");
           log::error(ex.what());
```
Comment (snoyer) [inline library/src/interactor_impl.cxx:452]:
```suggestion
          log::error("Interaction: error running command: \"" + commandWithArgs + "\": " + ex.what());
```
untested suggestion to improve spacing and line breaks, feel free to double check and/or tweak

**Pull request merged**